### PR TITLE
deps!: SC-13964 Update dependencies

### PR DESCRIPTION
**Pull request by alexregier**

### CODE REVIEW

**Review by Sese-Schneider — APPROVED** [review]

**Comment by alexregier** [comment]
@IchordeDionysos any idea on how to fix the node version issue?

**Comment by IchordeDionysos** [comment]
Not run it on Node 16? 🙈

**Comment by alexregier** [comment]
@IchordeDionysos can you help out here? I cannot update the node module version in the tsconfig to 18 because that value is not supported. But I want to avoid using commonjs for example.

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -17,7 +17,6 @@ jobs:
           - 'firebase-rules-coverage'
           - 'firebase-rules-generator'
         node_version:
-          - '16'
           - '18'
```
Comment (IchordeDionysos) [inline .github/workflows/checks.yaml:20]:
```suggestion
          - '18'
          - '20'
          - '22'
```

File: .github/workflows/checks.yaml [hunk 2]
```diff
@@ -41,7 +40,6 @@ jobs:
...
```
Comment (IchordeDionysos) [inline .github/workflows/checks.yaml:43]:
```suggestion
          - '18'
          - '20'
          - '22'
```

File: firebase-rules-coverage/yarn.lock [hunk 3]
```diff
@@ -1722,23 +1742,23 @@ graphemer@^1.4.0:
   integrity sha512-EtKwoO6kxCL9WO5xipiHTZlSzBm7WLT627TqC/uVRd0HKmq8NXyebnNYxDoBi7wt8eTWrUrKXCOVaFq9x1kgag==
 
 gts@^5.3.0:
-  version "5.3.0"
-  resolved "https://registry.yarnpkg.com/gts/-/gts-5.3.0.tgz#7adb72acd5789213efabf738301922cc30efd56d"
-  integrity sha512-/V0nbaWLBv8g0v2kol5M7Vf+kHXk19Ew5sa3wQJXeUaccesXg7AFo7eEInoIpR+aIJ3QDjoYt4zHYeFr8w8rng==
+  version "5.3.1"
```
Comment (IchordeDionysos) [inline firebase-rules-coverage/yarn.lock:1745]: We shouldn't update `gts` as documented in Confluence ☺️

**Comment by alexregier** [comment]
@IchordeDionysos The tests are complaining about `module` not being set to `node16`, but it is set to that value.

File: firebase-rules-coverage/package.json [hunk 4]
```diff
@@ -25,17 +25,17 @@
...
     "@jest/globals": "^29.7.0",
     "@types/jest": "^29.5.12",
     "@types/node": "^16.18.97",
-    "gts": "^5.3.0",
+    "gts": "5.3.0",
     "jest": "^29.7.0",
     "ts-jest": "^29.1.4",
     "ts-node": "^10.9.2",
-    "typescript": "~5.1.6"
+    "typescript": "~5.5.4"
```
Comment (plakhtiin) [inline firebase-rules-coverage/package.json:39]: @alexregier Have you tried <=5.4 version?

**Comment by richard-webb-dev** [comment]
i think it was this issue causing us grief:

https://github.com/kulshekhar/ts-jest/issues/4198

i also considered disabling this option that we were extending from google's tsconfig: https://www.typescriptlang.org/tsconfig/#noEmitOnError
it works, but i'm worried it would mask issues

other alternatives:
1. stick to TS5.1
2. move away from ts-jest
3. use cjs modules

tbh nothing was the perfect answer, but I'm not sure which compromise we want to have here.
I went with an updated typescript, and a funky jest.config.cjs, considering the project is small enough

File: firebase-rules-coverage/package.json (hunk 4 above)
Comment (richard-webb-dev) [reply firebase-rules-coverage/package.json:39]: see comment: https://github.com/simpleclub/firebase-rules-helper/pull/29#issuecomment-2315405180
5.1 would probably work, but 5.5 isn't the issue

**Review by richard-webb-dev — APPROVED** [review]

**Pull request merged**